NeurodataWithoutBorders/pynwb
# Add warnings when using positional arguments in container constructors

## Motivation

Fix #1968. Add warnings to `Container` constructor methods when using positional arguments. 

TODO
- [x] change PR target to merge into 3.0.0 branch for upcoming major release

## How to test the behavior?
```python
from pynwb import TimeSeries

TimeSeries('name', [10, 11, 12, 13], 'V', timestamps=[0, 1, 2, 3])
```

Output:
```
<stdin>:1: FutureWarning: TimeSeries.__init__: Using positional arguments for this method is discouraged and will be deprecated in a future major release. Please use keyword arguments to ensure future compatibility.
```

## Checklist

- [X] Is your contribution compliant with our coding style? This can be checked running `ruff check . && codespell` from the source directory.
